### PR TITLE
ARROW-723: [Python] Ensure that passing chunk_size=0 when writing Parquet file does not enter infinite loop

### PR DESCRIPTION
**Pull request by wesm**
This should also be fixed in parquet-cpp, will open a JIRA.

### CODE REVIEW

File: python/pyarrow/tests/test_parquet.py [hunk 1]
```diff
@@ -366,6 +366,20 @@ def test_multithreaded_read():
 
 
 @parquet
+def test_min_chunksize():
+    data = pd.DataFrame([np.arange(4)], columns=['A', 'B', 'C', 'D'])
+    table = pa.Table.from_pandas(data.reset_index())
+
+    buf = io.BytesIO()
+    pq.write_table(table, buf, chunk_size=0)
```
Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: What do you think about using -1 to mean min row group size and disallowing 0?

Comment (wesm) [inline python/pyarrow/tests/test_parquet.py]: i think the issue was the user did len(foo) / K and it truncated to zero. We could have it raise an exception instead

Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: Neither value makes sense logically, but -1 seems like a more obvious indicator that we're saying "do the default".

Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: Ah ok. Yeah I think raising an exception is probably the way to go here.

Comment (wesm) [inline python/pyarrow/tests/test_parquet.py]: chunk_size=None does the default also. I'll raise an exception for 0

Comment (cpcloud) [inline python/pyarrow/tests/test_parquet.py]: Yeah I saw that as well. I'm just going off intuition here, this shed is fully painted now 👍

**Review by cpcloud — APPROVED** [review]

**Pull request closed without merging**